firefox-ios/Client/Frontend/Autofill/LoginCellView.swift

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+import Common
+import Storage
+
+// MARK: - LoginCellView
+
+extension VerticalAlignment {
+    enum MidAccountAndName: AlignmentID {
+        static func defaultValue (in context: ViewDimensions) -> CGFloat {
+            context[.top]
+        }
+    }
+    static let midAccountAndName = VerticalAlignment(MidAccountAndName.self)
+}
+
+/// A view representing a cell displaying login information.
+struct LoginCellView: View {
+    // MARK: - Properties
+
+    @State private var textColor: Color = .clear
+    @State private var customLightGray: Color = .clear
+    @State private var iconPrimary: Color = .clear
+    @State private var stroke: Color = .clear
+
+    private(set) var login: EncryptedLogin
+    @Environment(\.themeType)
+    var theme
+    private(set) var onTap: () -> Void
+
+    // MARK: - Body
+
+    var body: some View {
+        Button(action: onTap) {
+            HStack(alignment: .midAccountAndName, spacing: 24) {
+                Image(StandardImageIdentifiers.Large.login)
+                    .renderingMode(.template)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 22)
+                    .padding(.leading, 16)
+                    .foregroundColor(iconPrimary)
+                    .alignmentGuide(.midAccountAndName) { $0[VerticalAlignment.center] }
+                VStack(alignment: .leading) {
+                    Text(login.decryptedUsername)
+                        .font(.body)
+                        .foregroundColor(textColor)
+                        .alignmentGuide(.midAccountAndName) { $0[VerticalAlignment.center] }
+                    Text("**********")
+                        .font(.subheadline)
+                        .foregroundColor(customLightGray)
+                }
+                Spacer()
+            }
+            .padding()
+        }
+        .buttonStyle(LoginButtonStyle(theme: theme.theme))
+        .listRowSeparator(.hidden)
+        .onAppear {
+            applyTheme(theme: theme.theme)
+        }
+        .onChange(of: theme) { newThemeValue in
+            applyTheme(theme: newThemeValue.theme)
+        }
+    }
+
+    // MARK: - Theme Application
+
+    /// Applies the theme to the view.
+    /// - Parameter theme: The theme to be applied.
+    func applyTheme(theme: Theme) {
+        let color = theme.colors
+        textColor = Color(color.textPrimary)
+        customLightGray = Color(color.textSecondary)
+        iconPrimary = Color(color.iconPrimary)
+        stroke = Color(color.actionSecondary)
+    }
+}
+
+// MARK: - CustomButtonStyle
+
+/// A login button style with a specific theme.
+struct LoginButtonStyle: ButtonStyle {
```

Comment:
Is that a button style that will be reused? If this is a button component style this potentially can go under the component library. CC @cyndichin

Reply:
It can in theory because we're supposed to have the same cells for all the autofill screen but figma has, and I don't joke, 3 different versions of this cell so that's why for now it's a LoginButtonStyle. In the future I expect it to become AutofillStyle?

Reply:
Designers should follow a standardized approach. We have components, Acorn and the design system being slowly put in place for those reasons. IMO the goal is that we have standardized cells and buttons, and not custom designs for one area of the application (in this case autofill). If that's not the case (this is what I understand from your comment?), you should discuss this with the designer as to what their plan is there.

Reply:
cc @jeffreygee @adudenamedruby

Reply:
@lmarceau Although I agree and from my last discussion we brought it up too @razvanlitianu if you remember. However this should not be a blocking change for this PR. 

Lets add a ticket for next part to address not just design as we also need to update JS component to use Autofill and not our old login scripts. So this work done is to support DMA in particular and reduce our sec risk (I can go into detail in DM) but there is a lot more work involved here in future.

Reply:
Yeah no it's not a blocker. I chatted with Razvan in DM over this too. It's just a process thing we should go over/pick up as another task if needed as this doesn't seem to align with how we want UI components to be built long term.

Reply:
100% agreed and thank you for bringing it up 🙏🏼 

@razvanlitianu I will merge this PR after BR run is good and lets use a future ticket to improve upon this